lnrpc/invoicesrpc/addinvoice.go

```diff
@@ -379,12 +380,21 @@ func AddInvoice(ctx context.Context, cfg *AddInvoiceConfig,
 		if len(openChannels) > 0 {
 			// We filter the channels by excluding the ones that were specified by
 			// the caller and were already added.
-			var filteredChannels []*channeldb.OpenChannel
+			var filteredChannels []*HopHintInfo
 			for _, c := range openChannels {
 				if _, ok := forcedHints[c.ShortChanID().ToUint64()]; ok {
 					continue
 				}
-				filteredChannels = append(filteredChannels, c)
+
+				hopHintInfo := newHopHintInfo(
+					c.ChannelFlags, c.FundingOutpoint,
```

Comment:
nit: since all these come from the passed `channeldb.OpenChannel` maybe `newHopHintInfo` could take that as an argument instead.